### PR TITLE
add lens function and tests

### PR DESCRIPTION
**Pull request by stoeffel**
How do you feel about adding a `lens` function to add more FP goodies to 101?

Checkout
* http://adit.io/posts/2013-07-22-lenses-in-pictures.html
* https://www.fpcomplete.com/school/to-infinity-and-beyond/pick-of-the-week/basic-lensing

### CODE REVIEW

**Comment by tjmehta** [comment]
Hey Stoeffel, I am a bit busy this weekend with my bro's bach party. Lens looks interesting going to read the links before I pull it

**Comment by stoeffel** [comment]
@tjmehta Np. Have fun at your bro' bach party.

**Comment by tjmehta** [comment]
This is going to go out with v1.0.0 this wkend give me a day or two

**Comment by stoeffel** [comment]
Cool :+1:

**Comment by tjmehta** [comment]
pulling this as I am behind schedule on the v1.0

File: lens.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  _lens.set = curry(setter, 2);
+  _lens.mod = curry(function(mod, obj) {
+    return _lens.set(mod(_lens(obj)), clone(obj));
+  });
+  return _lens;
+}
+
+function _setter(key) {
```
Comment (tjmehta) [inline lens.js]: use `101/put` ?

Comment (stoeffel) [inline lens.js]: I moved to `put`. Thanks for the hint.

**Comment by tjmehta** [comment]
:+1:

**Comment by tjmehta** [comment]
101@v0.16.0

**Pull request merged**